Source/WebCore/Modules/speech/SpeechSynthesis.cpp

```diff
@@ -151,6 +152,7 @@ void SpeechSynthesis::speak(SpeechSynthesisUtterance& utterance)
         return;
 #endif
 
+    utterance.setIsActiveForEventDispatch(true);
```

Comment:
oh, I meant that SpeechSynthesis would call makeActivity on utterance and store it there.
That way, you can clear that vector in addition to m_utteranceQueue and we're good to go.
But now I see that SpeechSynthesisUtterance has to clear this activity as well.
In that case, I think we can go back to your original approach. It's simpler & easier to follow.

Reply:
Oh. SpeechSynthesis does not uniquely own SpeechSynthesisUtterance (see `VTTCue::prepareToSpeak` for example), so clearing the vector won't make SpeechSynthesisUtterance go away.
Ok I will go with the previous approach as in https://github.com/WebKit/WebKit/commit/472814b941249f41ca2c1c89888008dfaa8ce3e6. Thanks for the review!